federation/pkg/dnsprovider/providers/google/clouddns/zones.go

```diff
@@ -30,12 +30,13 @@ type Zones struct {
 }
 
 func (zones Zones) List() ([]dnsprovider.Zone, error) {
+	var zoneList []dnsprovider.Zone
 	response, err := zones.impl.List(zones.project()).Do()
 	if err != nil {
-		return nil, err
+		return zoneList, nil
```

Comment:
Same comment here as above.  Only suppress the error if the call succeeded but returned an empty list.  For all other types of errors, you need to return them, not discard them here.